src/Compilers/CSharp/Test/Semantic/Semantics/NullableContextTests.cs

```diff
@@ -149,5 +152,218 @@ void AssertEnabledForInheritence(NullableContext context, bool warningsEnabled,
+            var expectedAnalyzedKeysAll = new[] { ".cctor", ".ctor", "F" };
+
+            verify(parseOptions: TestOptions.Regular, expectedFlowState: true, expectedAnalyzedKeysAll);
+            verify(parseOptions: TestOptions.Regular.WithFeature("nullableAnalysis", null), expectedFlowState: true, expectedAnalyzedKeysAll);
+            verify(parseOptions: TestOptions.Regular.WithFeature("nullableAnalysis", "true"), expectedFlowState: true, expectedAnalyzedKeysAll);
+            verify(parseOptions: TestOptions.Regular.WithFeature("nullableAnalysis", "false"), expectedFlowState: false);
+            verify(parseOptions: TestOptions.Regular.WithFeature("nullableAnalysis", "false").WithFeature("run-nullable-analysis", "false"), expectedFlowState: false);
+            verify(parseOptions: TestOptions.Regular.WithFeature("nullableAnalysis", "false").WithFeature("run-nullable-analysis", "true"), expectedFlowState: true, "F");
+            verify(parseOptions: TestOptions.Regular.WithFeature("nullableAnalysis", "true").WithFeature("run-nullable-analysis", "false"), expectedFlowState: false, expectedAnalyzedKeysAll);
+            verify(parseOptions: TestOptions.Regular.WithFeature("nullableAnalysis", "true").WithFeature("run-nullable-analysis", "true"), expectedFlowState: true, expectedAnalyzedKeysAll);
+
+            void verify(CSharpParseOptions parseOptions, bool expectedFlowState, params string[] expectedAnalyzedKeys)
+            {
+                var comp = CreateCompilation(source, parseOptions: parseOptions);
+                comp.NullableAnalysisData = new ConcurrentDictionary<object, int>();
+                if (expectedAnalyzedKeys.Length > 0)
+                {
+                    comp.VerifyDiagnostics(
+                        // (6,33): warning CS8603: Possible null reference return.
+                        //         if (obj == null) return null; // 1
+                        Diagnostic(ErrorCode.WRN_NullReferenceReturn, "null").WithLocation(6, 33));
+                }
+                else
+                {
+                    comp.VerifyDiagnostics();
+                }
+
+                var syntaxTree = comp.SyntaxTrees[0];
+                var model = comp.GetSemanticModel(syntaxTree);
+                var statement = syntaxTree.GetRoot().DescendantNodes().OfType<ReturnStatementSyntax>().Skip(1).Single();
+                Assert.Equal("return obj;", statement.ToString());
+                var typeInfo = model.GetTypeInfo(statement.Expression);
+                var expectedNullability = expectedFlowState ? Microsoft.CodeAnalysis.NullableFlowState.NotNull : Microsoft.CodeAnalysis.NullableFlowState.None;
+                Assert.Equal(expectedNullability, typeInfo.Nullability.FlowState);
+
+                var actualAnalyzedKeys = GetNullableDataKeysAsStrings(comp.NullableAnalysisData);
+                AssertEx.Equal(expectedAnalyzedKeys, actualAnalyzedKeys);
+            }
+        }
```

Comment:
>} [](start = 8, length = 1)

Do we have coverage for the code path that calls NullableWalker.AnalyzeAndRewrite in MethodCompiler.cs?

Reply:
This test hits that code path.

---
In reply to: [539752287](https://github.com/dotnet/roslyn/pull/49876#discussion_r539752287) [](ancestors = 539752287)

Reply:
How? This test doesn't appear to execute a mainline compile, but instead gets it's data from the semantic model. I wouldn't expect that to hit the MethodCompiler.

---
In reply to: [540378224](https://github.com/dotnet/roslyn/pull/49876#discussion_r540378224) [](ancestors = 540378224,539752287)

Reply:
Ah, apologies, this was on a different line in iteration 2 vs 7.

---
In reply to: [540606199](https://github.com/dotnet/roslyn/pull/49876#discussion_r540606199) [](ancestors = 540606199,540378224,539752287)